gordo_components/client/client.py

```diff
@@ -127,6 +128,34 @@ def __init__(
             ignore_unhealthy_targets=ignore_unhealthy_targets,
         )
 
+    @classmethod
+    @contextmanager
+    def cached_endpoints_client(cls):
+        """
+        Context manager for client, returns a modified version of :class:`Client`
+        which will only collect metadata from Watchman once. All subsequent instantiations
+        of this modified Client will use the previously loaded metadata/endpoints.
+
+        Examples
+        --------
+        >>>
+        >>> with Client.cached_endpoints_client() as CachedClient:  # doctest: +SKIP
+        ...     # Only the first instance will make a call to fetch metadata
+        ...     # subsequent instances will use the cached metadata in this block
+        ...     clients = [CachedClient(...) for i in range(100)]
+        """
+        setattr(cls, "__endpoints", None)
+        yield cls
+        delattr(cls, "__endpoints")
+
+    @classmethod
+    def _cache_endpoints(cls, endpoints: typing.List[EndpointMetadata]):
```

Comment:
Would it not make sense if "endpoints" was a map from `(host, port, scheme, gordo_version)` to the endpoints? As it is now it will get confused if you try to connect to several projects.